src/lib/containers/AccessApprovalsTable.tsx

```diff
@@ -0,0 +1,213 @@
+import moment from 'moment'
+import React, { useEffect, useState } from 'react'
+import { Table } from 'react-bootstrap'
+import { SMALL_USER_CARD } from '../utils/SynapseConstants'
+import { useSynapseContext } from '../utils/SynapseContext'
+import {
+  AccessApprovalSearchRequest,
+  AccessApprovalSearchResponse,
+  AccessApprovalSearchResult,
+} from '../utils/synapseTypes'
+import { accessApprovalSearch } from '../utils/SynapseClient'
+import Typography from '../utils/typography/Typography'
+import IconSvg from './IconSvg'
+import UserCard from './UserCard'
+import { displayToast } from './ToastMessage'
+import { PRODUCTION_ENDPOINT_CONFIG } from '../utils/functions/getEndpoint'
+import { SynapseSpinner } from './LoadingScreen'
+export type AccessApprovalsTableProps = {
+  accessorId: string
+  accessRequirementId?: string
+}
+export const AccessApprovalsTable: React.FunctionComponent<
+  AccessApprovalsTableProps
+> = ({ accessorId, accessRequirementId }) => {
+  const { accessToken } = useSynapseContext()
+  const [searchResult, setSearchResult] =
+    useState<AccessApprovalSearchResponse>()
+  const [isLoading, setIsLoading] = useState(false)
```

Comment:
Not necessary, but I think we could remove a lot of these state variables and simplify some logic if we delegate fetching this data to a custom `react-query` hook